src/site_scout/site_scout.py

```diff
@@ -140,6 +132,55 @@ def is_ascii(data: str) -> bool:
         """
         return all(ord(x) < 128 for x in data)
 
+    @classmethod
+    def create(
+        cls,
+        domain: str,
+        subdomain: str | None = None,
+        path: str = "/",
+        scheme: str = "http",
+    ) -> URL | None:
+        """Sanitize, verify data and create a URL if possible.
+
+        Args:
+            domain: Domain.
+            subdomain: Subdomain.
+            path: Path, must begin with '/'.
+            scheme: Scheme.
+
+        Returns:
+            URL object if input is valid otherwise None.
+        """
+        scheme = scheme.lower()
+        if scheme not in cls.ALLOWED_SCHEMES:
+            LOG.error("Cannot create URL: Invalid scheme %r", scheme)
+            return None
+
+        # use idna to encode domain with non ascii characters
+        domain = domain.lower().encode("idna").decode()
+        if not cls.VALID_DOMAIN.match(domain):
```

Comment:
Does `idna` encoding not guarantee this already?

Reply:
It would let though ASCII chars that are not supported in a domain such as $.